lib/ruby_http_client.rb

```diff
@@ -184,8 +184,8 @@ def build_args(args)
     #   - The final url string
     #
     def build_url(query_params: nil)
-      url = [add_version(''), *@url_path].join('/')
-      url = build_query_params(url, query_params) if query_params
+      url = [add_version(+''), *@url_path].join('/')
+      url = build_query_params(+url, query_params) if query_params
```

Comment:
I don't think this `url` is frozen because it is not “string literal” 🤔
```suggestion
      url = build_query_params(url, query_params) if query_params
```

Reply:
It is.  The line before does a join('/')

So it creates a string literal

Reply:
@childish-sambino, @rakatyal, @garethpaul, @tmimura39. Sorry to bother you but could look at this one?

Reply:
I agree with @tmimura39 - `url` is a _string_, but it's not a _string literal_. As proof:

```
irb(main):002> a = ["foo", "bar"]
=> ["foo", "bar"]
irb(main):003> a.map! { it << " world" }
(irb):3:in 'block in <top (required)>': can't modify frozen String: "foo" (FrozenError)
	from (irb):3:in 'Array#map!'
	from (irb):3:in '<main>'
	from <internal:kernel>:168:in 'Kernel#loop'
	from /Users/chrishowlett/.local/share/mise/installs/ruby/3.4.1/lib/ruby/gems/3.4.0/gems/irb-1.15.1/exe/irb:9:in '<top (required)>'
	from /Users/chrishowlett/.local/share/mise/installs/ruby/3.4.1/bin/irb:25:in 'Kernel#load'
	from /Users/chrishowlett/.local/share/mise/installs/ruby/3.4.1/bin/irb:25:in '<main>'
irb(main):004> a.join("/") << "world"
=> "foo/barworld"
```

Another bump for getting this merged - we depend on this gem and it's blocking our ability to enable frozen literals in 3.4, and will (probably) Just Break from Ruby 4

Reply:
You're right! Idk why I thought it was back then. There are so many recent methods that create frozen strings 

For example "false.to_s" creates a frozen string

Reply:
Pushed. Thanks